### ISSUE TITLE
$ARG1 doesn't work in subprocess mode

### ISSUE THREAD

**Issue by battaglia01** [xonsh/xonsh#2718]
It seems like $ARGS works in both subprocess and python mode, but naming a specific argument such as $ARG1 does not. For example:

```
$ cat a.xsh
#!/usr/bin/env xonsh

echo $ARGS
echo $ARG1

print($ARGS)
print($ARG1)

$ ./a.xsh 1 2 3 4
['./a.xsh', '1', '2', '3', '4']
$ARG1
['./a.xsh', '1', '2', '3', '4']
1
```

xonfig output:
```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.6.8                |
| Git SHA          | a88a485f             |
| Commit Date      | Jun 28 20:10:30 2018 |
| Python           | 3.5.2                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | 1.0.15               |
| shell type       | prompt_toolkit       |
| pygments         | None                 |
| on posix         | True                 |
| on linux         | True                 |
| distro           | neon                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

**Comment by battaglia01** [comment]
Happens as well upon upgrading to 0.7.0:

```
+------------------+----------------------+
| xonsh            | 0.7.0                |
| Git SHA          | 3779e8b5             |
| Commit Date      | Jul 14 14:40:04 2018 |
| Python           | 3.5.2                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | 1.0.15               |
| shell type       | prompt_toolkit1      |
| pygments         | None                 |
| on posix         | True                 |
| on linux         | True                 |
| distro           | neon                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

**Comment by scopatz** [comment]
Potential fix in #2739
